.github/workflows/publish.yml

```diff
@@ -32,8 +35,9 @@ jobs:
           tags: |
             type=sha,format=long
             type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
-            type=raw,value=${{env.PYTHON_FLASK_DEV_VANILLA}},enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
-            type=raw,value=${{env.PYTHON_FLASK_DEV_FRONTEND}},enable=(${{ (github.ref == format('refs/heads/{0}', 'main')) && (matrix.flavour == 'frontend') }})
+            type=raw,value=${{env.PYTHON_FLASK_DEV_PYTHON}},enable=${{ github.ref == format('refs/heads/{0}', 'main')}}
+            type=raw,value=${{env.PYTHON_FLASK_DEV_FLASK}},enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=raw,value=${{env.PYTHON_FLASK_DEV_FRONTEND}},enable=${{ (github.ref == format('refs/heads/{0}', 'main')) && (matrix.flavour == 'frontend') }}
             type=ref,event=branch
```

Comment:
It will tag them differently if it's not on main, but basically if it's building the `flask` image it will tag with the flask tag, if it's building frontend, it will tag with both frontend and flask. Happy to change if it makes no sense when we start using it